Sources/Algorithms/Rotate.swift

```diff
@@ -132,15 +132,17 @@ extension MutableCollection {
   /// - Returns: The new index of the element that was at the start of
   ///   `subrange` pre-rotation.
   ///
-  /// - Complexity: O(*n*), where *n* is the length of `subrange`.
+  /// - Complexity: O(*n*), where *n* is the length of `subrange` relative to
+  ///   this collection.
   @inlinable
   @discardableResult
-  public mutating func rotate(
-    subrange: Range<Index>,
+  public mutating func rotate<Range: RangeExpression>(
+    subrange: Range,
     toStartAt newStart: Index
-  ) -> Index {
-    var m = newStart, s = subrange.lowerBound
-    let e = subrange.upperBound
+  ) -> Index where Range.Bound == Index {
+    let cSubrange = subrange.relative(to: self)
```

Comment:
Question: what does the "c" in `cSubrange` mean?

Reply:
I really don't remember sorry, but changed to `rSubrange` for `relative` as it makes more sense I guess.

Reply:
Perhaps `relativeSubrange` will help everyone understand?

Reply:
Just fixed